javascript/code-challenges/tree/binarySearchTree.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+let Node = require('./node');
+
+class BinarySearchTree {
+  constructor() {
+    this.root = null;
+  }
+
+  add(value) {
+    if (this.root === null) {
+      this.root = new Node(value);
+    } else {
+      if (this.root.left === null) {
+        this.root.left = new Node(value)
+      } else {
+        if (this.root.right === null) {
+          this.root.right = new Node(value)
+        }
+      }
+    }
+  }
+}
+module.exports = BinarySearchTree;
```

Comment:
Looks like you are off to a great start! Once you have some time to come back to this to add Pre and In to your BinarySearch and then contains to your BinarySearchTree and tests for those methods I will regrade for you!